producer-kafka/README.adoc

```diff
@@ -0,0 +1,407 @@
+To see how to achieve it, please visit the link:https://docs.confluent.io/current/schema-registry/avro.html#compatibility-types[Schema Registry Guide].
+- There are two Kafka producer settings which cannot be set by the user: `key.serializer` and `value.serializer` - both must be set to `KafkaAvroSerializer.class`
+because the CDC producer uses Avro format for publishing events to Kafka. Any value provided for these two settings will be ignored.
+
+
+=== Metrics Settings
+The metrics framework has three settings that allow you to control it:
+
+- `cdc.kafka.metrics.enabled` - whether the CDC extension should register all Kafka producer metrics in the `MetricRegistry`. This setting is required. It should contain a boolean value.
+- `cdc.kafka.metrics.include-tags` - whether the registered metric name should include tags(if exists). This setting is optional. If not specified, it will be set to `true`. It should contain a boolean value.
+- `cdc.kafka.metrics.name` - this is the prefix used to expose metrics on the `MetricRegistry`. This setting is optional. If not specified, it will be set to `producer`.
+The CDC connector also adds its own prefix to every metric: `cdc.kafka`. If you leave this setting unset, every metric will be prefixed with `cdc.kafka.producer.`
+
+== Schema Creation
+When the `createTableSchemaAsync()` is invoked on the CDC producer, it constructs the Avro `Schema` for key and value.
+Let's take a look at the process of constructing a Avro schema.
+
+=== Key Schema
+The key schema contains all partition keys from the TableMetadata passed to the `createTableSchemaAsync`.
+For example, if the table metadata contains one primary key with a field name: `pk_1` of a string type, the constructed schema will be:
+[source,json]
+----
+{
+   "type":"record",
+   "name":"Key",
+   "namespace":"topicPrefix.keyspaceName.tableName",
+   "fields":[
+      {
+         "name":"pk_1",
+         "type":"string"
+      }
+   ]
+}
+----
+Please note that this schema's name will be `Key`, but this schema is created in the context of a specific namespace.
+The namespace equals a topic name constructed based on the `cdc.kafka.topic.prefix-name` and `TableMetadata` keyspace and table.
+Once the schema for a specific key is created, you cannot remove or rename any existing field. Such a change will not be backward compatible.
+By doing so, you will break the consumers of the Kafka topic. You are allowed to add a new Primary Key because adding a new field to an Avro schema is backward compatible.
+
+
+=== Value Schema
+The value schema is more complicated because it must allow three states for any non-Primary Key field.
+The first state is an actual value denoting that this value changed. The second state is the absence of a value, meaning no change for a given field.
+The third state that it should allow is null value for a given field. This denotes clearing of the specific value (due to either being deleted or set to null explicitly).
```

Comment:
I think these paragraphs can be confusing...

We can express it as what happens and how is that represented in the Kafka schema, for example:

> During a database mutation, a given column of a row can be given a new value, it can be cleared (set to `null`) or left unchanged. In the same way, for a given field the `Value` schema can contain the new value, the `null` value or the absence of the value.